independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/ContextInstancesGenerator.java

```diff
@@ -75,105 +78,128 @@ Collection<Resource> generate(DotName scope) {
                 .setModifiers(ACC_PUBLIC);
-        // this.lockAll();
         // ContextInstanceHandle<?> copy1 = this.1;
-        // this.unlockAll();
         // Set<ContextInstanceHandle<?>> ret = new HashSet<>();
         // if (copy1 != null) {
         //    ret.add(copy1);
         // }
         // return ret;
-        getAllPresent.invokeVirtualMethod(MethodDescriptor.ofMethod(contextInstances.getClassName(), "lockAll", void.class),
-                getAllPresent.getThis());
         List<ResultHandle> results = new ArrayList<>(idToFields.size());
         for (InstanceAndLock fields : idToFields.values()) {
             results.add(getAllPresent.readInstanceField(fields.instance, getAllPresent.getThis()));
         }
-        getAllPresent.invokeVirtualMethod(MethodDescriptor.ofMethod(contextInstances.getClassName(), "unlockAll", void.class),
-                getAllPresent.getThis());
         ResultHandle ret = getAllPresent.newInstance(MethodDescriptor.ofConstructor(HashSet.class));
         for (ResultHandle result : results) {
             getAllPresent.ifNotNull(result).trueBranch().invokeInterfaceMethod(MethodDescriptors.SET_ADD, ret, result);
         }
         getAllPresent.returnValue(ret);
     }
 
-    private void implementLockAll(ClassCreator contextInstances, Map<String, InstanceAndLock> idToFields) {
-        MethodCreator lockAll = contextInstances.getMethodCreator("lockAll", void.class)
-                .setModifiers(ACC_PRIVATE);
-        for (InstanceAndLock fields : idToFields.values()) {
-            ResultHandle lock = lockAll.readInstanceField(fields.lock, lockAll.getThis());
-            lockAll.invokeInterfaceMethod(MethodDescriptors.LOCK_LOCK, lock);
-        }
-        lockAll.returnVoid();
-    }
-
-    private void implementUnlockAll(ClassCreator contextInstances, Map<String, InstanceAndLock> idToFields) {
-        MethodCreator unlockAll = contextInstances.getMethodCreator("unlockAll", void.class)
-                .setModifiers(ACC_PRIVATE);
-        for (InstanceAndLock fields : idToFields.values()) {
-            ResultHandle lock = unlockAll.readInstanceField(fields.lock, unlockAll.getThis());
-            unlockAll.invokeInterfaceMethod(MethodDescriptors.LOCK_UNLOCK, lock);
-        }
-        unlockAll.returnVoid();
-    }
-
-    private void implementRemoveEach(ClassCreator contextInstances, Map<String, InstanceAndLock> idToFields) {
+    private void implementRemoveEach(ClassCreator contextInstances, List<MethodDescriptor> removeInstances) {
         MethodCreator removeEach = contextInstances.getMethodCreator("removeEach", void.class, Consumer.class)
                 .setModifiers(ACC_PUBLIC);
-        // this.lockAll();
-        // ContextInstanceHandle<?> copy1 = this.1;
-        // if (copy1 != null) {
-        //    this.1 = null;
-        // }
-        // this.unlockAll();
+        // ContextInstanceHandle<?> copy1 = remove1();
```

Comment:
Hm, so if we remove `lockAll()` then this operation is not atomic (in the sense of the set of context instances we operate on), right? And that's probably ok, because it's only used during the destruction of a context.